network/netcore/src/transport/tcp.rs

```diff
@@ -88,18 +87,22 @@ impl Transport for TcpTransport {
         Ok((
             TcpListenerStream {
                 inner: listener,
-                config,
+                config: self.clone(),
             },
             actual_addr,
         ))
     }
 
     fn dial(&self, addr: NetworkAddress) -> Result<Self::Outbound, Self::Error> {
-        let socket_addr = multiaddr_to_string(&addr)?;
-        let config = self.clone();
+        let (hostport, _addr_suffix) =
+            parse_hostport(addr.as_slice()).ok_or_else(|| invalid_addr_error(&addr))?;
         let f: Pin<Box<dyn Future<Output = io::Result<TcpStream>> + Send + 'static>> =
-            Box::pin(TcpStream::connect(socket_addr));
-        Ok(TcpOutbound { inner: f, config })
+            Box::pin(TcpStream::connect(hostport));
```

Comment:
(hostport makes it sound like it's just a port)

Reply:
sock_addr maybe?